src/server/pkg/storage/fileset/util.go

```diff
@@ -56,3 +65,20 @@ func WriteTarStream(ctx context.Context, w io.Writer, fs FileSource) error {
 	}
 	return tar.NewWriter(w).Close()
 }
+
+// CleanTarPath ensures that the path is in the canonical format for tar header names.
+// This includes ensuring a prepending /'s and ensure directory paths
+// have a trailing slash.
+func CleanTarPath(x string, isDir bool) string {
+	y := "/" + strings.Trim(x, "/")
```

Comment:
This is what `cleanPath` does right? Maybe `cleanPath` should be in the fileset package and exported, then used by this function?

Reply:
Paths in the driver have to be the same whether they are directories or files, so that matching works.  We cannot call this function without `isDir` which we don't have at the usual call sites of `cleanPath`

Reply:
I meant line 73 specifically. I think we could have a `CleanPath` function exposed by the fileset package that would get called at line 73.

Reply:
I'm not sure that would get reuse in the driver. I think we may find that it makes the path matching in the driver simpler to work on paths without the leading slash.  I don't think there is a good reason to create a dependency here yet.